src/overview.js

```diff
@@ -0,0 +1,385 @@
+        this.logger.debug('Processing card ' + card.name);
+        let actions = this.getActions(card, lists[card.idList]);
+        for (const [name, action] of actions) {
+          if (projects.has(name)) {
+            if (!projects.get(name).actions.has(board.name)) {
+              projects.get(name).actions.set(board.name, new Map());
+            }
+            projects.get(name).actions.get(board.name).set(action.name, action);
+          }
+        }
+      }
+    }
+  };
+
+  /**
+   * Add an action checklist item.
+   */
+  this.addCheckItem = async (checklistId, action) => {
+    try {
+      await this.trelloClient.post('/checklists/' + checklistId + '/checkItems', {
+        name: action.name,
+        checked: action.complete
+      });
+      this.logger.debug('Added checkitem ' + action.name);
+    }
+    catch (exception) {
+      this.logger.error('Unable to add checkitem ' + action.name + ' to list ' + checklistId);
+    }
+  };
+
+  /**
+   * Update an action checklist item.
+   */
+  this.updateCheckItem = async (cardId, checklistId, id, action) => {
+    try {
+      await this.trelloClient.post('/cards/' + cardId + '/checklist/' + checklistId + '/checkItem/' + id, {
+        idChecklistCurrent:  checklistId,
+        idCheckItem:  id,
+        name:  action.name,
+        state:  action.complete,
+      });
+      this.logger.debug('Updated checkitem ' + id);
+    }
+    catch (exception) {
+      this.logger.error('Unable to update checkitem ' + id);
+    }
+  };
+
+  /**
+   * Delete an action checklist item.
+   */
+  this.deleteCheckItem = async (checklistId, id) => {
+    try {
+      await this.trelloClient.delete('/checklists/' + checklistId + '/checkItems/' + id, {
+        idCheckItem:  id,
+      });
+      this.logger.debug('Deleted checkitem ' + id);
+    }
+    catch (exception) {
+      this.logger.error('Unable to delete checkitem ' + id);
+    }
+  };
+
+  /**
+   * Add a board checklist.
+   */
+  this.addChecklist = async (cardId, name, actions) => {
+    try {
+      const checklist = await this.trelloClient.post('/checklists', {
+        idCard:  cardId,
+        name:  name,
+      });
+      this.logger.debug('Added checklist ' + name);
+
+      // Add all actions.
+      for (const action of actions.values()) {
+        this.logger.debug('Processing action ' + action.name);
+        await this.addCheckItem(checklist.id, action);
+      }
+    }
+    catch (exception) {
+      this.logger.error('Unable to add checklist ' + name + ' to card ' + cardId);
+    }
+  };
+
+  /**
+   * Update board checklist: add, update or remove actions.
+   */
+  this.updateChecklist = async (cardId, checklist, actions) => {
+    let items = new Map();
+    for (const action of actions.values()) {
+      this.logger.debug('Processing action ' + action.name);
+      items.set(action.name, action);
+    }
+
+    for (const checkitem of checklist.checkItems) {
+      if (items.has(checkitem.name)) {
```

Comment:
This should extract the link from the check list item name ("link - does - status - due") to do the lookup in the `items` map.

`action.name` may indeed differ from `checkitem.name` (previous value), for example if the status, doer or due changed. Only the beginning with the link is the same and serves as identifier.

Ref: https://github.com/UN-OCHA/legacy--rw-trello-overview/blob/fe4a507e53bf6d3fc28c58c5d3c064e5020bfa65/main.go#L231-L238

Reply:
Changed, but :100: % not sure